plugins/modules/ec2_key.py

```diff
@@ -173,6 +195,10 @@ def extract_key_data(key):
         data['tags'] = boto3_tag_list_to_ansible_dict(key['Tags'])
     if 'KeyMaterial' in key:
         data['private_key'] = key['KeyMaterial']
+    if 'KeyType' in key:
+        data['type'] = key['KeyType']
+    elif key_type:
+        data['type'] = key_type
```

Comment:
could this return the wrong thing if someone passed 'key_type' on an existing key (which then wouldn't get created and would be the original type)?

Reply:
if the module is called on an existing key pair but the key_type is different from the original one, the key pair will be recreated with the new key_type